app/models/threshold_condition.rb

```diff
@@ -2,27 +2,21 @@
 
 # ThresholdCondition
 class ThresholdCondition < Condition
+  # When someone answers that they are 'experiencing symptoms' and does not
+  # fill out a complete report, we use this function to generate a list of
+  # symptoms with nil values to build a ReportedCondition with nil values
+  # flagging the monitoree for manual follow-up.
   def clone_symptoms_remove_values
-    new_symptoms = []
-    symptoms.each do |symptom|
-      new_symptom = symptom.dup
-      new_symptom.value = nil
-      new_symptoms.push(new_symptom)
-    end
-    new_symptoms
+    new_symptoms = symptoms.deep_dup
```

Comment:
Doing this worries me because the symptoms in new_symptoms after you do symptoms.deep_dup all have the same ids as what were in symptoms. Whereas the way they the cloning done before, the resulting symptoms in new_symptoms had nil ids (to be populated on save). I don't like that this way keeps the old ids because it would be very bad if the reported conditions  symptoms overwrote the threshold conditions symptoms. In the console I ran `deep_duped = ThresholdCondition.last.symptoms.deep_dup` and also ran the old code that built new_symptoms individually. with these results. 
![image](https://user-images.githubusercontent.com/3967953/81945242-d86ad080-95cb-11ea-8a2b-31d1655dfb19.png)

Notice how the symptoms in deep_duped has an ID and populated created_at and updated_at fields, whereas what was in new_symptoms has all of the info to create a new record with the identical name and label values only?

This same logic applies to the changes made in the `clone_symptoms_negate_bool_values` function too

Reply:
Please see my updates. One thing that is present in both cases is that the `condition_id` will be preserved which represents an associated condition for the symptom..

Reply:
@mayerm94 I pushed a new assertion that proves that the id is not set after manipulation.